### PR TITLE
RSValue array cleanup - [MOD-7145]

### PR DESCRIPTION
**Pull request by GuyAv46**
**Describe the changes in the pull request**

Remove unnecessary initialization and branching, as we almost exclusively were calling `RSValue_NewArrayEx(arr, len, RSVAL_ARRAY_NOINCREF | RSVAL_ARRAY_ALLOC);` (with these flags), and in the few cases we don't we can get the same behavior with minor adjustments.
We can then delete `RSValue_NewArrayEx` and use `RSValue_NewArray` that gives the same behavior as the prior call with the commonly used flags, without branching and no-op loops

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/aggregate/reducers/to_list.c [hunk 1]
```diff
@@ -53,16 +53,16 @@ static RSValue *tolistFinalize(Reducer *rbase, void *ctx) {
   TrieMapIterator *it = TrieMap_Iterate(tlc->values, "", 0);
   char *c;
   tm_len_t l;
-  void *ptr;
-  RSValue **arr = rm_calloc(tlc->values->cardinality, sizeof(RSValue));
+  RSValue *val;
+  RSValue **arr = rm_malloc(tlc->values->cardinality * sizeof(RSValue*));
```
Comment (DvirDukhan) [inline src/aggregate/reducers/to_list.c:57]: use `rm_calloc`
also, for cleaner code I suggest we'll have a dedicated function for this instead of direct call to `rm_malloc`

File: src/aggregate/reducers/random_sample.c [hunk 2]
```diff
@@ -21,7 +21,7 @@ static void *sampleNewInstance(Reducer *base) {
   size_t blocksize = MAX(10000, sizeof(rsmplCtx) + r->len * sizeof(RSValue *));
   rsmplCtx *ctx = Reducer_BlkAlloc(base, sizeof(*ctx) + r->len * sizeof(RSValue *), blocksize);
   ctx->seen = 0;
-  ctx->samplesArray = RSValue_NewArrayEx(NULL, r->len, 0);
+  ctx->samplesArray = RSValue_NewArray(rm_malloc(sizeof(RSValue *) * r->len), 0);
```
Comment (DvirDukhan) [inline src/aggregate/reducers/random_sample.c:24]: why did you change `NULL` to new allocation?

Comment (GuyAv46) [reply src/aggregate/reducers/random_sample.c:24]: To preserve the behavior. Passing the requested length with no values will set the array length to 0 and allocate this amount of memory for future use

File: src/aggregate/reducers/to_list.c (hunk 1 above)
Comment (DvirDukhan) [reply src/aggregate/reducers/to_list.c:57]: we have bad experience with un-initilized memory :)

The function should be something like
```c
RSValue **RSVvalue_NewArray(size_t count) {
    return rm_calloc(count, sizeof(RSValue);
}
```

Let's try to avoid low-level calls where we can abstract them

File: src/value.h [hunk 3]
```diff
@@ -368,6 +368,13 @@ RSValue *RS_Int64Val(int64_t ii);
  */
 RSValue *RSValue_NewArray(RSValue **vals, uint32_t len);
 
+/**
+ * Helper function to allocate memory before passing it to RSValue_NewArray
+ */
+static inline RSValue **RSValue_AllocateArray(uint32_t len) {
+  return (RSValue **)rm_malloc(len * sizeof(RSValue *));
```
Comment (DvirDukhan) [inline src/value.h:375]: calloc :)

Comment (GuyAv46) [reply src/value.h:375]: Don’t we care about the performance difference? In all the places we fill the entire array anyway. This is not a partially initialized struct but an array of pointers.
I added a `memset` call in the only place we might not fill the array (and in this case it also most likely a but in the trieMap or its iterator

**Review by DvirDukhan — APPROVED** [review]
